### PR TITLE
Block reconsumption of response stream

### PR DESCRIPTION
**Pull request by timowestnosto**

### CODE REVIEW

File: framework/src/play/libs/ws/WSAsync.java [hunk 1]
```diff
@@ -663,6 +667,43 @@ public InputStream getStream() {
             }
         }
 
+        @Override
+        public Document getXml(String encoding) {
+            try {
+                checkConsumed();
+                return super.getXml(encoding);
+            } finally {
+                consumed = true;
```
Comment (leandroutn) [inline framework/src/play/libs/ws/WSAsync.java]: are you sure every time `super.getXml()` throws an exception we should mark it as `consumed`?

Comment (leandroutn) [inline framework/src/play/libs/ws/WSAsync.java]: talked about this and I now agree this looks good

**Comment by leandroutn** [comment]
looks good to me

File: framework/src/play/libs/ws/WSAsync.java (hunk 1 above)
Comment (timowestnosto) [inline framework/src/play/libs/ws/WSAsync.java]: the idea is to allow getString() and getXml() calls only once, on exceptions it's probable that the stream is already partially consumed.

**Pull request merged**